### ISSUE TITLE
Error swallowed if address in use

### ISSUE THREAD

**Issue by alanshaw** [libp2p/js-libp2p#311]
I already have a node running - if I try to start another then the process exits and no error is printed. I have to use `DEBUG` to see the error.

```console
$ DEBUG=* npm start

> custom-ipfs-repo@0.1.0 start /Users/alan/Code/protocol-labs/js-ipfs/examples/custom-ipfs-repo
> node index.js

  jsipfs booting +0ms
  repo opening at: /tmp/custom-repo/.ipfs +0ms
  repo init check +2ms
  repo:version comparing version: 7 and 7 +0ms
  repo init null { config: true, spec: true, version: undefined } +2ms
  repo:lock locking /tmp/custom-repo/.ipfs/repo.lock +0ms
  repo aquired repo.lock +4ms
  repo creating datastore +0ms
  repo creating blocks +0ms
  repo creating keystore +1ms
  repo all opened +0ms
  jsipfs:state -> initializing +0ms
  jsipfs init +10ms
  jsipfs pre-start +1ms
  jsipfs no keychain, use --pass +0ms
  jsipfs peer created +13ms
  jsipfs Loaded pins from the datastore +12ms
  jsipfs:state -> stopped +27ms
  jsipfs starting +1ms
  jsipfs:state -> starting +0ms
  libp2p:switch:dialer create: 8 peer limit, 30000 dial timeout +0ms
  libp2p:connection-manager options: {"maxPeers":null,"minPeers":0,"maxData":null,"maxSentData":null,"maxReceivedData":null,"maxEventLoopDelay":null,"pollInterval":2000,"movingAverageInterval":60000,"defaultPeerValue":1,"maxPeersPerProtocol":{}} +0ms
  libp2p libp2p is starting +0ms
  libp2p:switch:transport adding TCP +0ms
  libp2p:switch:transport adding WebSockets +1ms
  latency-monitor:LatencyMonitor latencyCheckIntervalMs: 500 dataEmitIntervalMs: 2000 +0ms
  latency-monitor:LatencyMonitor Expecting ~0.25 events per summary +1ms
  latency-monitor:LatencyMonitor Using process.hrtime for timing +0ms
  latency-monitor:LatencyMonitor localData: { deltaOffset: 491, startTime: [ 88806, 946617583 ] } +0ms
  libp2p:switch The switch is starting +0ms
  libp2p:tcp:listen Listening on 4002 0.0.0.0 +0ms
  libp2p:switch:error Error: listen EADDRINUSE: address already in use 0.0.0.0:4002
  libp2p:switch:error     at Server.setupListenHandle [as _listen2] (net.js:1290:14)
  libp2p:switch:error     at listenInCluster (net.js:1338:12)
  libp2p:switch:error     at doListen (net.js:1471:7)
  libp2p:switch:error     at process._tickCallback (internal/process/next_tick.js:63:19) +0ms
  libp2p:error Error: listen EADDRINUSE: address already in use 0.0.0.0:4002
  libp2p:error     at Server.setupListenHandle [as _listen2] (net.js:1290:14)
  libp2p:error     at listenInCluster (net.js:1338:12)
  libp2p:error     at doListen (net.js:1471:7)
  libp2p:error     at process._tickCallback (internal/process/next_tick.js:63:19) +0ms
```

**Comment by jacobheun** [comment]
This doesn't appear to be an issue with libp2p, but rather js-ipfs.  I ran the code from js-ipfs master and attempted to spawn the daemon twice. The first run `node ./src/cli/bin.js daemon` worked as expected, but the second run fails silently, but the failure is due to the repo already being locked

```log
$ DEBUG=* node ./src/cli/bin.js daemon
Initializing IPFS daemon...
  jsipfs:http-api starting +0ms
  jsipfs:http-api failed to load optional electron-webrtc dependency +0ms
  jsipfs:http-api failed to load optional webrtc dependency +1ms
  jsipfs booting +0ms
  repo opening at: /Users/jacobheun/.jsipfs +0ms
  repo init check +7ms
  repo:version comparing version: 7 and 7 +0ms
  repo init null { config: true, spec: true, version: undefined } +11ms
  repo:lock locking /Users/jacobheun/.jsipfs/repo.lock +0ms
  jsipfs:http-api error starting core { Error: Lock file is already being hold
    at options.fs.stat (/Users/jacobheun/git/ipfs/js-ipfs/node_modules/proper-lockfile/lib/lockfile.js:54:47)
    at /Users/jacobheun/git/ipfs/js-ipfs/node_modules/graceful-fs/polyfills.js:285:20
    at FSReqWrap.oncomplete (fs.js:155:5) code: 'ELOCKED', file: '/Users/jacobheun/.jsipfs' } +74ms
  jsipfs:http-api done { Error: Lock file is already being hold
    at options.fs.stat (/Users/jacobheun/git/ipfs/js-ipfs/node_modules/proper-lockfile/lib/lockfile.js:54:47)
    at /Users/jacobheun/git/ipfs/js-ipfs/node_modules/graceful-fs/polyfills.js:285:20
    at FSReqWrap.oncomplete (fs.js:155:5) code: 'ENOENT', file: '/Users/jacobheun/.jsipfs' } +1ms
  repo:lock Error: Lock file is already being hold
  repo:lock     at options.fs.stat (/Users/jacobheun/git/ipfs/js-ipfs/node_modules/proper-lockfile/lib/lockfile.js:54:47)
  repo:lock     at /Users/jacobheun/git/ipfs/js-ipfs/node_modules/graceful-fs/polyfills.js:285:20
  repo:lock     at FSReqWrap.oncomplete (fs.js:155:5) +14ms
```

**Comment by jacobheun** [comment]
libp2p start/stop not erroring resolved by libp2p/js-libp2p#316 and error swallowing in js-ipfs is slated to be resolved by https://github.com/ipfs/js-ipfs/pull/1860
